### PR TITLE
Dynamic bucket size

### PR DESCRIPTION
**Pull request by cnderrauber**
Reject packet size exceeds MaxPktSize(1500)

### CODE REVIEW

**Review by davidzhao — APPROVED** [review]
nice!

File: pkg/bucket/bucket.go [hunk 1]
```diff
@@ -29,33 +28,56 @@ const (
...
-func (b *Bucket) ResyncOnNextPacket() {
-	b.resyncOnNextPacket = true
+// Grow increases the capacity of the bucket by adding initial capacity to the buffer
+func (b *Bucket) Grow() int {
+	pktSize := MaxPktSize + pktSizeHeader
+	buf := make([]byte, pktSize*b.initCapacity)
+	newSlots := make([][]byte, b.initCapacity)
+	for i := 0; i < b.initCapacity; i++ {
+		newSlots[i] = buf[i*pktSize : (i+1)*pktSize]
+		binary.BigEndian.PutUint16(newSlots[i], invalidPktSize)
```
Comment (boks1971) [inline pkg/bucket/bucket.go:64]: Do this in the above constructor. Or just make a function to pass in the buf, slots and do it there?

File: pkg/bucket/bucket.go [hunk 2]
```diff
@@ -29,33 +28,56 @@ const (
...
 	init               bool
 	resyncOnNextPacket bool
 	step               int
 	headSN             uint16
+	initCapacity       int
 	maxSteps           int
 }
 
-func NewBucket(buf *[]byte) *Bucket {
+func NewBucket(capacity int) *Bucket {
```
Comment (boks1971) [inline pkg/bucket/bucket.go:40]: probably add `maxCapacity` as a safeguard?

**Review by boks1971 — APPROVED** [review]

File: pkg/bucket/bucket.go (hunk 2 above)
Comment (cnderrauber) [reply pkg/bucket/bucket.go:40]: The bucket will not increase itself, all capacity is controlled by buffers.

**Pull request merged**